src/mono/wasm/debugger/BrowserDebugProxy/MonoSDBHelper.cs

```diff
@@ -1540,11 +1540,21 @@ public async Task<JObject> InvokeMethod(ArraySegment<byte> valueTypeBuffer, int
             return await CreateJObjectForVariableValue(retDebuggerCmdReader, varName, false, -1, false, token);
         }
 
-        public async Task<JObject> InvokeMethodInObject(int objectId, int methodId, string varName, CancellationToken token)
+        public async Task<JObject> InvokeMethodInObject(int objectId, int methodId, ElementType elementType, string varName, CancellationToken token)
         {
-            using var commandParamsObjWriter = new MonoBinaryWriter();
-            commandParamsObjWriter.Write(ElementType.Class, objectId);
-            return await InvokeMethod(commandParamsObjWriter.GetParameterBuffer(), methodId, varName, token);
+            switch (elementType)
+            {
+                case ElementType.ValueType:
+                {
+                    return await InvokeMethod(valueTypes[objectId]?.valueTypeBuffer, methodId, varName, token);
```

Comment:
Instead of adding the element type to the id, why not just check if `valueTypes[objectId] != null`? If it is, then it is a valuetype, else call it like an object.

Reply:
Are you sure that objectId is unique, regardless of the schema? I just tried to base the decision on the 
```
if (valueTypes[objectId] == null) -> return class path result
otherwise -> return valueType path result
```
and such a change broke many tests, including: EvaluateExpressionsWithElementAccessNested, EvaluateExpressionsWithElementAccessByLocalVariable, EvaluateExpressionsWithElementAccessMultidimentional, EvaluateExpressionsWithElementAccessByConstant etc.

Reply:
Looking at the code, the id for valuetype is generated by `var valueTypeId = Interlocked.Increment(ref debuggerObjectId);` in `CreateJObjectForValueType`, so it should be unique. The only other type using `debuggerObjectId` is pointers. Objects are getting the objectId from sdb.

Reply:
Sorry, it was my mistake - failures were caused by accessing the dict value by valueType[objectId] instead of doing TryGetValue.